client/admin.go

```diff
@@ -46,23 +54,140 @@ func NewAdminClientWithCredentials(address, username, password string, opts ...O
 		basicAuth = basicAuth.Insecure()
 	}
 
-	return &grpcAdminClient{client: svcv1.NewCerbosAdminServiceClient(grpcConn), creds: basicAuth}, nil
+	return &GrpcAdminClient{client: svcv1.NewCerbosAdminServiceClient(grpcConn), creds: basicAuth}, nil
 }
 
-type grpcAdminClient struct {
+type GrpcAdminClient struct {
 	client svcv1.CerbosAdminServiceClient
 	creds  credentials.PerRPCCredentials
 }
 
-func (gac *grpcAdminClient) AddOrUpdatePolicy(ctx context.Context, policies *PolicySet) error {
+func (c *GrpcAdminClient) AddOrUpdatePolicy(ctx context.Context, policies *PolicySet) error {
 	if err := policies.Validate(); err != nil {
 		return err
 	}
 
 	req := &requestv1.AddOrUpdatePolicyRequest{Policies: policies.policies}
-	if _, err := gac.client.AddOrUpdatePolicy(ctx, req, grpc.PerRPCCredentials(gac.creds)); err != nil {
+	if _, err := c.client.AddOrUpdatePolicy(ctx, req, grpc.PerRPCCredentials(c.creds)); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+type recvFn func() (*responsev1.ListAuditLogEntriesResponse, error)
+
+// collectLogs collects logs from the receiver function and passes to the channel
+// it will return an error if the channel type is not accepted.
+func collectLogs(receiver recvFn, channel interface{}) error {
```

Comment:
I see, this whole magic was derived form having two methods for this :) My ultimate goal was to have most simplistic methods it could have. As you suggested, these are admin API methods so we may overlook the simplicity here. Going to update accordingly.